### PR TITLE
Mkl invoke

### PR DESCRIPTION
**Pull request by agibsonccc**
Adds mkl invocation

### CODE REVIEW

File: include/loops/indexreduce.h [hunk 1]
```diff
@@ -192,9 +191,9 @@ template<typename OpType>
 		sPartials[threadIdx.x] = OpType::startingIndexValue(dx);
 
 		//length for the tad
-		__shared__ volatile Nd4jIndex xLength;
```
Comment (saudet) [inline include/loops/indexreduce.h]: Didn't @raver119 change that recently?

File: blas/CMakeLists.txt [hunk 2]
```diff
@@ -18,6 +18,7 @@ if(WIN32)
     endforeach()
 endif()
 
+find_package(MKL)
```
Comment (saudet) [inline blas/CMakeLists.txt]: libnd4j doesn't link with BLAS or MKL right now, so we don't need that...

Comment (agibsonccc) [inline blas/CMakeLists.txt]: I had to add this to get it to use mkl. If it didn't change anything I wouldn't have opened the PR.

Comment (saudet) [inline blas/CMakeLists.txt]: Well anyway, adding this doesn't actually do anything, the variables are not used

**Pull request closed without merging**